niworkflows/viz/utils.py

```diff
@@ -503,6 +503,9 @@ def plot_melodic_components(melodic_dir, in_file, tr=None,
     in_nii = nb.load(in_file)
     if not tr:
         tr = in_nii.header.get_zooms()[3]
+        units = in_nii.header.get_xyzt_units()
+        if units and units[-1] == 'msec':
+            tr = tr/1000.0
```

Comment:
Is it actually possible for `units` to evaluate to `False`? Do we want to handle `'usec'` units, for completeness? And should we warn on unknown units? It's almost always seconds, but the warning might help us track down the issue, if it's causing one.

So I might rewrite as:

```Python
_, t_unit = in_nii.header.get_xyzt_units()
if t_unit == 'unknown':
    warnings.warn('Unknown t units, assuming seconds.')
    t_unit = 'sec'
tr /= 10.0 ** {'sec': 0, 'msec': 3, 'usec': 6}[t_unit]
```

Would also be okay to just put spaces around `/`.